dask/array/routines.py

```diff
@@ -160,6 +161,57 @@ def dot(a, b):
     return tensordot(a, b, axes=((a.ndim - 1,), (b.ndim - 2,)))
 
 
+def _inner_apply_along_axis(arr,
+                            func1d,
+                            func1d_axis,
+                            func1d_args,
+                            func1d_kwargs):
+    return np.apply_along_axis(
+        func1d, func1d_axis, arr, *func1d_args, **func1d_kwargs
+    )
+
+
+@wraps(np.apply_along_axis)
+def apply_along_axis(func1d, axis, arr, *args, **kwargs):
+    arr = asarray(arr)
+
+    # Validate and normalize axis.
+    arr.shape[axis]
+    axis = len(arr.shape[:axis])
+
+    # Test out some data with the function.
+    test_data = np.ones((1,), dtype=arr.dtype)
+    test_result = np.array(func1d(test_data, *args, **kwargs))
+
+    if (LooseVersion(np.__version__) < LooseVersion("1.13.0") and
+            (np.array(test_result.shape) > 1).sum() > 1):
+            raise ValueError(
+                "Only one non-trivial dimension allowed in result. "
+                "Need NumPy 1.13.0+ for this functionality."
```

Comment:
Made some small tweaks to this error message in PR ( https://github.com/dask/dask/pull/2703 ).